hedera-node/docs/design/services/token-service/airdrops/airdrop-transaction.md

```diff
@@ -0,0 +1,205 @@
+ *    The fee for this transfer SHALL include the transfer, the association, the cost to renew
+ *    that association once, the airdrop fee, and any custom fees. If the pending airdrop is not
+ *    claimed immediately, the `sender` SHALL pay the cost to renew the token association, and
+ *    the cost to maintain the pending airdrop, until the pending airdrop is claimed or cancelled.
+ *  - An account with no available automatic association slots SHALL have a "Pending Airdrop"
+ *    created and must claim that airdrop with a `claimAirdrop` transaction.<br/>
+ *    The fee for this transfer SHALL include the transfer, the association, the cost to renew
+ *    that association once, the airdrop fee, and any custom fees. If the pending airdrop is not
+ *    claimed immediately, the `sender` SHALL pay the cost to renew the token association, and
+ *    the cost to maintain the pending airdrop, until the pending airdrop is claimed or cancelled.
+ *
+ * If an airdrop would create a pending airdrop for a fungible/common token, and a pending airdrop
+ * for the same sender, receiver, and token already exists, the existing pending airdrop
+ * SHALL be updated to add the new amount to the existing airdrop, rather than creating a new
+ * pending airdrop.
+ *
+ * Any airdrop that completes immediately SHALL be irreversible. Any airdrop that results in a
+ * "Pending Airdrop" MAY be canceled via a `cancelAirdrop` transaction.
+ *
+ * All transfer fees (including custom fees and royalties), as well as the rent cost for the
+ * first auto-renewal period for any automatic-association slot occupied by the airdropped
+ * tokens, SHALL be charged to the account paying for this transaction.
+ *
+ * ### Record Stream Effects
+ * - Each successful transfer SHALL populate that transfer in `token_transfer_list` for the record.
+ * - Each successful transfer that consumes an automatic association slot SHALL populate the
+ *   `automatic_association` field for the record.
+ * - Each pending transfer _created_ SHALL be added to the `pending_airdrops` field for the record.
+ * - Each pending transfer _updated_ SHALL be added to the `pending_airdrops` field for the record.
+ */
+ message TokenAirdropTransactionBody {
+/**
+     * A list of token transfers representing one or more airdrops.<br/>
+     * The sender for each transfer MUST have sufficient balance to complete the transfers.
+     * <p>
+     * All token transfers MUST successfully transfer tokens or create a pending airdrop
+     * for this transaction to succeed.<br/>
+     * This list MUST contain between 1 and 10 transfers, inclusive.
+     * <p>
+     * <blockquote>Note that each transfer of fungible/common tokens requires both a debit and
+     * a credit, so each _fungible_ token transfer MUST have _balanced_ entries in the
+     * TokenTransferList for that transfer.</blockquote>
+     */
+     repeated TokenTransferList token_transfers= 1;
```

Comment:
Using a `TokenTransferList` here doesn't seem right. How are we doing to enforce there is one credit and one debit in each transfer ? 

Seems like we need a different protobuf definition that has one credit and one debit.
Allowing user to set multiple credits and debits will make it confusing to create a `PendingAirdropId` that needs one sender and one receiver

Reply:
@tinker-michaelj @netopyr @Nana-EC   What do you think ?

Reply:
Each Pending Airdrop is created individually from a _single_ transfer in the list, and only if _necessary_. The majority of transfers are expected to complete without creating a pending airdrop.
I would expect to see one PendingAirdropId for _each_ combination of sender and recipient in the transfer list(s), and _only_ if the recipient cannot immediately receive the airdrop.

Reply:
In the case of fungible airdrops having the same sender and receiver - we would result in a single transfer with the fungible value aggregated.

In the case of NFT airdrops with different serial numbers but the same sender and receiver - we would have a single transfer in the list for each separate serial number with the same sender and receiver.

This is my understanding, having the current HIP definition in mind.

Reply:
I still don't understand, I think we will need to add validations in code. But probably don't want to block this PR. 🙂